File: voting-system-message-broker/src/main/java/com/voting/ConfigReader.java
package com.voting;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class ConfigReader {
	private String filename;
	private String messageBrokerIp;
	private int messageBrokerPort;
    private String messageBrokerUser;
    private String messageBrokerPwd;
    private String messageBrokerVirtualHost;
    private String exchangeName;
	
    public ConfigReader(String filename) throws FileNotFoundException{
    	this.filename = filename;
    	this.loadConfig();
    }

    public String getMessageBrokerIp() {
    	return this.messageBrokerIp;
    }
    
    public int getMessageBrokerPort() {
    	return this.messageBrokerPort;
    }
    
    public String getMessageBrokerUser() {
    	return this.messageBrokerUser;
    }
    
    public String getMessageBrokerPwd() {
    	return this.messageBrokerPwd;
    }
    
    public String getMessageBrokerVirtualHost() {
    	return this.messageBrokerVirtualHost;
    }
    
    public String getExchangeName() {
    	return this.exchangeName;
    }
    private void loadConfig() throws FileNotFoundException {
    	File file = new File(this.filename);
    	
    	messageBrokerIp = loadMessageBrokerIp(file);
    	messageBrokerPort = loadMessageBrokerPort(file);
    	messageBrokerUser = loadMessageBrokerUser(file);
    	messageBrokerPwd = loadMessageBrokerPwd(file);
    	messageBrokerVirtualHost = loadMessageBrokerVirtualHost(file);
    	exchangeName = loadExchangeName(file);
    }
    
    private String loadMessageBrokerIp(File file) throws FileNotFoundException {
    	return loadConfValue(file, "messageBrokerIp:");
    }
    
    private int loadMessageBrokerPort(File file) throws FileNotFoundException {
    	return Integer.parseInt(loadConfValue(file, "messageBrokerPort:"));
    }
    
    private String loadMessageBrokerUser(File file) throws FileNotFoundException {
    	return loadConfValue(file, "messageBrokerUser:");
    }
    
    private String loadMessageBrokerPwd(File file) throws FileNotFoundException {
    	return loadConfValue(file, "messageBrokerPwd:");
    }
    
    private String loadMessageBrokerVirtualHost(File file) throws FileNotFoundException {
    	return loadConfValue(file, "messageBrokerVirtualHost:");
    }
    
    private String loadExchangeName(File file) throws FileNotFoundException {
    	return loadConfValue(file, "exchangeName:");
    }
    
    private String loadConfValue(File file, String beggining) throws FileNotFoundException {
    	Scanner reader = new Scanner(file);
    	while (reader.hasNextLine()) {
            String data = reader.nextLine();
            if(data.startsWith(beggining)) {
                reader.close();
            	return getValueAfterColon(data);
            }
    	}
        reader.close();
    	throw new IllegalArgumentException(beggining + " not found in given config file");
    }
    
    private String getValueAfterColon(String line) {
    	return line.substring(line.lastIndexOf(":") + 1).replaceAll("\\s+","");
    }
}

File: voting-system-visualizer/configReader.py
import string


class ConfigReader():

    def __init__(self, filename):
        self.filename = filename
        self.load_config()

    def load_config(self):
        file = open(self.filename, "r")
        self.message_broker_ip = self.load_message_broker_ip(file)
        self.message_broker_port = self.load_message_broker_port(file)
        self.message_broker_user = self.load_message_broker_user(file)
        self.message_broker_pwd = self.load_message_broker_pwd(file)
        self.message_broker_virtual_host = self.load_message_broker_virtual_host(file)
        self.exchange_name = self.load_exchange_name(file)
        file.close()

    def load_message_broker_ip(self, file):
        return self.load_conf_value(file, "messageBrokerIp:")

    def load_message_broker_port(self, file):
        return self.load_conf_value(file, "messageBrokerPort:")

    def load_message_broker_user(self, file):
        return self.load_conf_value(file, "messageBrokerUser:")

    def load_message_broker_pwd(self, file):
        return self.load_conf_value(file, "messageBrokerPwd:")

    def load_message_broker_virtual_host(self, file):
        return self.load_conf_value(file, "messageBrokerVirtualHost:")

    def load_exchange_name(self, file):
        return self.load_conf_value(file, "exchangeName:")

    def load_conf_value(self, file, beggining):
        for line in file:
            if line.startswith(beggining):
                non_whitespace_line = line.translate({ord(c): None for c in string.whitespace})
                return non_whitespace_line[len(beggining):]
        print('No ' + beggining + ' found in given conf file')
        return ''